saleor/attribute/models/base.py

```diff
@@ -214,6 +214,120 @@ def get_translated_keys(self):
         return {"name": self.name}
 
 
+class AttributeValueManager(models.Manager):
+    def _prepare_query_for_bulk_operation(self, objects_data):
+        query_params = models.Q()
+
+        for obj in objects_data:
+            defaults = obj.pop("defaults")
+            query_params |= models.Q(**obj)
+            obj["defaults"] = defaults
+
+        return self.filter(query_params)
+
+    def _is_correct_record(self, record, obj):
+        is_correct_record = (
+            getattr(record, field_name) == field_value
+            for field_name, field_value in obj.items()
+            if field_name != "defaults"
+        )
+        return all(is_correct_record)
+
+    def bulk_get_or_create(self, objects_data):
+        # this method mimics django's queryset.get_or_create method on bulk objects
+        # instead of performing it one by one
+        # https://docs.djangoproject.com/en/5.0/ref/models/querysets/#get-or-create
+
+        results = []
+        objects_not_in_db: List[AttributeValue] = []
+
+        # prepare a list that will save order index of attribute values
+        objects_enumerated = list(enumerate(objects_data))
+        query = self._prepare_query_for_bulk_operation(objects_data)
+
+        # iterate over all records in db and check if they match any of objects data
+        for record in query.iterator():
+            # iterate over all objects data and check if they match any of records in db
+            for index, obj in objects_enumerated:
+                if self._is_correct_record(record, obj):
+                    # upon finding existing record add it to results
+                    results.append((index, record))
+                    # remove it from objects list, so it won't be added to new records
+                    objects_enumerated.remove((index, obj))
+
+                    break
+
+        # add what is left to the list of new records
+        self._add_new_records(objects_enumerated, objects_not_in_db, results)
+        # sort results by index as db record order might be different from sort_order
+        results.sort()
+        results = [obj for index, obj in results]
+
+        if objects_not_in_db:
+            self.bulk_create(objects_not_in_db)  # type: ignore[arg-type]
```

Comment:
I might be missing something or not fully familiar with the details but:
What will happen if you aren't able to create the objects in DB? For example, one already exists in DB. 
Django's `get_or_create` makes `.get` once again in that case